pep-0517.txt

```diff
@@ -218,6 +241,14 @@ metadata. The directory passed in by the build frontend MUST be
 identical to the directory created by ``get_wheel_metadata``,
 including any unrecognized files it created.
 
+This method only places the *contents* of the wheel file, without actually
+creating the final artifact. This is to allow flexibility in how the result is
```

Comment:
Basically doing it that way means that in some cases we're going to round trip through zip and then unzip for no real reason. In your example, flit is going to produce a zipped up wheel, then pip is going to immediately unzip it. However if flit is modified to stage the files to an immediate directory, then using this API pip can just install directly from that, instead of taking the performance hit of a compress + decompress round trip.